### DISCUSSION TITLE
Idea to improve the songselect filters like: bpm=120 cs<5

### DISCUSSION THREAD

**Discussion by Inconsist** [ppy/osu#22307] [Ideas] [answered]
I recently saw the new designs for the songselect and thought that maybe you couldnt just have a fader for the star rating but for other parameters too.

quick visual suggestion:
![Screenshot 2023-01-20 210256](https://user-images.githubusercontent.com/87947227/213813357-108c5de7-2c85-48b7-939c-0810bacc8c69.png)

#18890 is where this design is originally from btw.

This way you could get rid of the old typed out filters, which can be cluttered if you want to define multiple ranges and sometimes straight out not work. Things like "status!=r" still dont work...(btw why, status=r works fine)

I think nobody would really want to type "bpm>180 bpm<200 cs<5 ar>9.0 ar<9.9 hp<5 status=r" if the same thing could be done more visually intuitive and with less googling and potential errors for users which are new to the filter options and quicker for those who already know them. And you can always still type the filters out manually if you desire so.

Faders could be a very elagant solutuion to this and are probably more intuitive for the majority of users I imagine.
you could implement this a dropdown thing like modselect is currently, so it wouldnt take up alot of space.(my example is incomplete its just to visualize the idea in genreal)
Im happy to hear your feedback on this idea, maybe we can integrate it in the new UI design at some point.

**Comment by Inconsist** [comment]
Oh and to make things like CS==5.5 more easy with the faders, maybe a toogle for absolute values vs ranges...

**Comment by bdach** [comment, marked as answer]
https://github.com/ppy/osu/discussions/17494

**Reply by Inconsist** [reply]
Btw if you need anyone to help design this, Im not good at coding but design is something I know alot more about.
